lib/cloudant.js

```diff
@@ -822,28 +834,38 @@ Cloudant.prototype.replaceOrCreate = function replaceOrCreate(
   var self = this;
   var idName = self.idName(model);
   var mo = self.selectModel(model);
+  var id = data[idName].toString();
+
+  // Callback handler for both create calls.
+  var createHandler = function(err, id) {
+    if (err) return cb(err);
+    mo.db.get(id, function(err, doc) {
+      if (err) return cb(err);
+      cb(err, self.fromDB(model, mo, doc), {isNewInstance: true});
+    });
+  };
 
-  var id = data[idName];
   if (id) {
-    self.getCurrentRevision(model, id, function(err, rev) {
+    self.exists(model, id, {}, function(err, count) {
       if (err) return cb(err);
-      if (rev) data._rev = rev;
-      self._insert(model, data, function(err) {
-        if (err) return cb(err);
-        mo.db.get(id, function(err, doc) {
+      else if (count > 0) {
+        self.getCurrentRevision(model, id, function(err, rev) {
```

Comment:
:-1:

Again, this function defeats the purpose of my PR. It only works because it's hiding how Cloudant should work.